federatedscope/autotune/fedex/client.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+from federatedscope.core.message import Message
+from federatedscope.core.worker import Client
+
+logger = logging.getLogger(__name__)
+
+
+class FedExClient(Client):
+    """Some code snippets are borrowed from the open-sourced FedEx (https://github.com/mkhodak/FedEx)
+    """
+    def _apply_hyperparams(self, hyperparams):
+        """Apply the given hyperparameters
+        Arguments:
+            hyperparams (list): each element is a dict, where keys are hyperparameter names and values are specific choices.
+        """
+
+        cmd_args = []
+        for hyper in hyperparams:
+            for k, v in hyper.items():
+                cmd_args.append(k)
+                cmd_args.append(v)
+
+        self._cfg.defrost()
+        self._cfg.merge_from_list(cmd_args)
+        self._cfg.freeze()
+
+        self.trainer.ctx.setup_vars()
```

Comment:
@DavdGao hi dawei, please help me validate whether this line can actually update the hyper-parameters of the client. thx!

Reply:
Executing the code in line 24-26 will change the attribute `self.trainer.ctx.cfg` and calling function `setup_vars` will update the hyper-parameters and partial variables. However, the variables like `model`, `data`, `device` in `self.trainer.ctx` won't change as they are already created within `self.trainer.ctx`.